src/workers/filesystems/infrastructure/SyncRemoteFileSystem.ts

```diff
@@ -0,0 +1,666 @@
+                file.encrypt_version
+              ) +
+              (file.type ? `.${file.type}` : ''),
+            file,
+          }))
+          .filter(({ name }) => {
+            const isValid = fileNameIsValid(name);
+
+            if (!isValid) {
+              Logger.warn(
+                `REMOTE file with name ${name} will be ignored due an invalid name`
+              );
+              return false;
+            }
+
+            return true;
+          })
+          .forEach(({ file, name }) => {
+            const modificationTime = getSecondsFromDateString(
+              file.modificationTime
+            );
+            listing[name] = RemoteItemMetaData.from({
+              modtime: modificationTime,
+              size: file.size,
+              isFolder: false,
+              id: file.id,
+            });
+            cache[name] = {
+              id: file.id,
+              parentId: file.folderId,
+              isFolder: false,
+              bucket: file.bucket,
+              fileId: file.fileId,
+              modificationTime,
+              size: file.size,
+            };
+          });
+
+        foldersInThisFolder.forEach((folder) => {
+          const name =
+            currentName +
+            crypt.decryptName(
+              folder.name,
+              (folder.parent_id as number).toString(),
+              '03-aes'
+            );
+          listing[name] = RemoteItemMetaData.from({
+            modtime: getSecondsFromDateString(folder.updated_at),
+            size: 0,
+            isFolder: true,
+            id: folder.id,
+          });
+          cache[name] = {
+            id: folder.id,
+            parentId: folder.parent_id as number,
+            isFolder: true,
+            bucket: folder.bucket,
+          };
+          traverse(folder.id, `${name}/`);
+        });
+      }
+
+      return { listing, readingMetaErrors };
+    },
+
+    async deleteFile(name: string): Promise<void> {
+      const fileInCache = cache[name];
+
+      const result = await clients.newDrive.post(
+        `${process.env.NEW_DRIVE_URL}/drive/storage/trash/add`,
+        {
+          items: [
+            {
+              type: 'file',
+              id: fileInCache.fileId,
+            },
+          ],
+        }
+      );
+
+      if (result.status !== 200) {
+        await handleFetchError(
+          result.data(),
+          'Moving remote file to trash',
+          `Name: ${name}, fileInCache: ${JSON.stringify(fileInCache, null, 2)}`
+        );
+      }
+    },
+
+    async renameFile(oldName: string, newName: string): Promise<void> {
+      const fileInCache = cache[oldName];
+      const newNameBase = path.parse(newName).name;
+
+      try {
+        const res = await httpRequest(
+          `${process.env.API_URL}/api/storage/file/${fileInCache.fileId}/meta`,
+          {
+            method: 'POST',
+            headers: { ...headers, 'internxt-mnemonic': mnemonic },
```

Comment:
Definitely remove the header 'internxt-mnemonic'

Reply:
Removed